### ISSUE TITLE
Support for Kerberos connection

### ISSUE THREAD

**Issue by feenst** [exposebox/node-thrift2-hbase#11]
I'm evaluating [node-hbase](https://github.com/adaltas/node-hbase) for connecting to an HBase instance, but it's using REST rather than Thrift to connect.

Has Kerberos support for node-thirft2-hbase been discussed?

**Comment by moshewe** [comment]
We haven't discussed Kerberos support, nor do I expect this to get into our
development plans any time soon...

You can, however, easily swap the connection pool we have now with a
Kerberos connection pool. You are more than welcome to fork and submit a
PR, and we'll be glad to help.

**Comment by feenst** [comment]
@moshewe, I'm not sure where to start on swapping out the connection pool. Would that be something [here](https://github.com/exposebox/node-thrift2-hbase/blob/master/src/client.js#L45)?

**Comment by moshewe** [comment]
Yes. You can see that in the `create` callback given to the pool it creates
a `Client` object. You can change that to a Kerberos client connection.

**Comment by Ran-P** [comment]
In client.js line 147 the connection to the thrift server is created.
The `createConnection` is in the node-thrift package, It just create a stream using `net.createConnection`.
You can add your function of `createKerberosConnection` :

```javascript
thrift.createKerberosConnection = function(host, port, options) {
  var stream = kerberos.connect(port, host, options);
  var connection = new Connection(stream, options);
  connection.host = host;
  connection.port = port;
  return connection;
};
```

**Comment by feenst** [comment]
@Ran-P, the Node kerberos packages I've seen ([kerberos](https://github.com/mongodb-js/kerberos) and [node-krb5](https://github.com/adaltas/node-krb5)) both seem to work by generating a token that's used with an `Authentication` HTTP header.

The thrift connection uses sockets / streams. Any idea how Kerberos authentication works in that scenario?

**Comment by moshewe** [comment]
As @Ran-P pointed out, you should discuss this with the `node-thrift` people. If you figure something out with them, then you can see exactly where @Ran-P had already directed you to how to use it with our package - it's just a library for us.

**Comment by Ran-P** [comment]
@feenst you can use httpConnection in the thrift library instead of the RPC connection.
Add(or extend) to the httpConnection to use krb5.
